### PR TITLE
libspng: add maintainer, fix dependencies, args

### PR DESCRIPTION
**Pull request by sethrj**

### CODE REVIEW

File: var/spack/repos/builtin/packages/libspng/package.py [hunk 1]
```diff
@@ -14,10 +14,17 @@ class Libspng(CMakePackage):
...
-    depends_on("c", type="build")  # generated
-    depends_on("cxx", type="build")  # generated
+    depends_on("c", type="build")
+    depends_on("zlib")
+
+    def cmake_args(self):
+        target = self.spec.target
+        args = [self.define("BUILD_EXAMPLES", False)]
+        args.append(self.define("ENABLE_OPT", target.vendor != "generic"))
+        return args
```
Comment (alecbcs) [inline var/spack/repos/builtin/packages/libspng/package.py]:
```suggestion
    def cmake_args(self):
        target = self.spec.target
        return [ 
            self.define("BUILD_EXAMPLES", False)),
            self.define("ENABLE_OPT", target.vendor != "generic"),
        ]
```

Quick suggestion to clean up the syntax a bit.

**Review by alecbcs — COMMENTED** [review]
One quick suggestion, otherwise this PR looks good to me. Thanks @sethrj

File: var/spack/repos/builtin/packages/libspng/package.py (hunk 1 above)
Comment (sethrj) [reply var/spack/repos/builtin/packages/libspng/package.py]: Yep thanks, I'd originally planned to put that second statement inside an `if` block (silly me).

**Comment by sethrj** [comment]
@spackbot fix style

**Comment by sethrj** [comment]
Ping @alecbcs

File: var/spack/repos/builtin/packages/libspng/package.py [hunk 2]
```diff
@@ -14,10 +14,18 @@ class Libspng(CMakePackage):
     homepage = "https://github.com/randy408/libspng"
     url = "https://github.com/randy408/libspng/archive/refs/tags/v0.7.2.tar.gz"
 
+    maintainers("sethrj")
     license("BSD-2-Clause")
```
Comment (alecbcs) [inline var/spack/repos/builtin/packages/libspng/package.py]:
```suggestion
    maintainers("sethrj")

    license("BSD-2-Clause")
```

Quick suggestion to add a space between `maintainers()` and `license()` directives as described in the packaging guide.

**Review by alecbcs — APPROVED** [review]
Looks good to me. Thanks @sethrj!

**Pull request merged**